src/python_testing/TestSpecParsingSupport.py

```diff
@@ -106,6 +221,80 @@ def test_write_optional(self):
             asserts.assert_equal(xml_cluster.attributes[ATTRIBUTE_ID].write_optional,
                                  write_support == 'optional', "Unexpected write_optional value")
 
+    def test_derived_clusters(self):
+        clusters: dict[int, XmlCluster] = {}
+        derived_clusters: dict[str, XmlCluster] = {}
+        ids_by_name: dict[str, int] = {}
+        problems: list[ProblemNotice] = []
+        base_cluster_xml = ElementTree.fromstring(BASE_CLUSTER_XML_STR)
+        derived_cluster_xml = ElementTree.fromstring(DERIVED_CLUSTER_XML_STR)
+        expected_global_attrs = [GlobalAttributeIds.FEATURE_MAP_ID, GlobalAttributeIds.ATTRIBUTE_LIST_ID,
+                                 GlobalAttributeIds.ACCEPTED_COMMAND_LIST_ID, GlobalAttributeIds.GENERATED_COMMAND_LIST_ID, GlobalAttributeIds.CLUSTER_REVISION_ID]
+
+        add_cluster_data_from_xml(base_cluster_xml, clusters, derived_clusters, ids_by_name, problems)
+        add_cluster_data_from_xml(derived_cluster_xml, clusters, derived_clusters, ids_by_name, problems)
+
+        asserts.assert_equal(len(clusters), 1, "Unexpected number of clusters")
+        asserts.assert_equal(len(derived_clusters), 1, "Unexpected number of derived clusters")
+        asserts.assert_equal(len(ids_by_name), 1, "Unexpected number of IDs per name")
+        asserts.assert_equal(len(problems), 0, "Unexpected number of problems")
+        asserts.assert_equal(ids_by_name["Test Derived"], 0xFFFF, "Test derived name not added to IDs")
+
+        asserts.assert_true(0xFFFF in clusters, "Derived ID not found in clusters")
+        asserts.assert_equal(set(clusters[0xFFFF].attributes.keys()), set(
+            [0, 2, 3] + expected_global_attrs), "Unexpected attribute list")
+        asserts.assert_equal(set(clusters[0xFFFF].accepted_commands.keys()), set([]), "Unexpected accepted commands")
+        asserts.assert_equal(set(clusters[0xFFFF].generated_commands.keys()), set([]), "Unexpected generated commands")
+
+        asserts.assert_true("Test Base" in derived_clusters, "Base ID not found in derived clusters")
+        asserts.assert_equal(set(derived_clusters["Test Base"].attributes.keys()), set(
+            [0, 1, 2, 3] + expected_global_attrs), "Unexpected attribute list")
+        asserts.assert_equal(set(derived_clusters["Test Base"].accepted_commands.keys()), set([0]), "Unexpected accepted commands")
+        asserts.assert_equal(set(derived_clusters["Test Base"].generated_commands.keys()),
+                             set([1]), "Unexpected generated commands")
+        asserts.assert_equal(str(derived_clusters["Test Base"].accepted_commands[0].conformance),
+                             "M", "Unexpected conformance on base accepted command")
```

Comment:
For the error message is this unexpected conformance on `base` or `derived`?

Reply:
I think the confusion here is around the naming - I ran with the old naming and really the dict should be called "pure_base_clusters". Will change.